### ISSUE TITLE
Anonymous function using const

### ISSUE THREAD

**Issue by km147** [javascript-obfuscator/javascript-obfuscator#659]
Won’t store an anonymous function using const. Changed const to var and it works. There are a lot of uses of const like that in our code but didn’t like that particular usage!

Example: socket['on']('extraParam', async _0x2455a8 => {
        if (_0x2455a8['type'] === 'candidate' && _0x2455a8['payload']) const _0x4ed198 = setInterval(async () => {

Yeah there’s nothing wrong with that line!

Storing an anonymous function to a constant could be a perceived problem as it will run once and never run again if being strict
But in this situation that's exactly what we want it to do and that’s the purpose of using const there - js obfuscator breaks it

**Comment by km147** [comment]
All test conducted with Simplify enabled +
Compact Code
then
Various other requirements we generally select including string array, debug, disable console. All gave the same result. 

Further work..
Its actually seems to be a problem with an async anonymous function as a callback from setInterval. The following doesn't work:

if (true) {
    const interval = setInterval(async () => {
        //...
    },500)
}

But it works without the if statement or by changing the const to var

**Comment by sanex3339** [comment]
Please show me the full example that works without obfuscation and doesn't work after obfuscation, because your code looks ok and I don't see any problems here:
```
if (true) {
const interval = setInterval(async () => {
//...
},500)
}
```

**Comment by sanex3339** [comment]
Without a full example, i can't help.

For example, it could be a code that returns some data in the console. And after obfuscation, it won't.

**Comment by sanex3339** [comment]
Looks like this is the same as https://github.com/javascript-obfuscator/javascript-obfuscator/issues/661. I'll close this issue in favor of https://github.com/javascript-obfuscator/javascript-obfuscator/issues/661

**Comment by sanex3339** [comment]
Released as `1.5.1`
